src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/EventSubscriber/ReadProductsEventSubscriberSpec.php

```diff
@@ -43,15 +47,22 @@ public function it_provides_subscribed_events(): void
         $this->getSubscribedEvents()->shouldReturn([ReadProductsEvent::class => 'saveReadProducts',]);
     }
 
-    public function it_saves_read_products_events(
+    public function it_saves_read_products_events_with_a_rest_api_event(
         $connectionContext,
-        $updateDataDestinationProductEventCountHandler
+        $updateDataDestinationProductEventCountHandler,
+        Connection $connection,
+        ConnectionCode $connectionCode
     ): void {
-        $connection = new Connection('ecommerce', 'ecommerce', FlowType::DATA_DESTINATION, 42, 10);
-        $connectionContext->getConnection()->willReturn($connection);
-        $connectionContext->isCollectable()->willReturn(true);
+        $readProductsEvent = new ReadProductsEvent(3);
+        $connection->hasDataDestinationFlowType()->willReturn(true)->shouldBeCalledTimes(1);
```

Comment:
```suggestion
        $connection->hasDataDestinationFlowType()->willReturn(true);
```
No matter if it is called or not, that's not the purpose of the test.
Same remark for all the lines after.

The only `->should` you must have is this one https://github.com/akeneo/pim-community-dev/pull/14503/files#diff-88334017883f497d59648ee8218b74001a064d2e5224d0337d3d0b29b5432d51R67-R73